ProjectLighthouse/Helpers/CensorHelper.cs

```diff
@@ -18,6 +20,67 @@ public static class CensorHelper
         "UwU", "OwO", "uwu", "owo", "o3o", ">.>", "*pounces on you*", "*boops*", "*baps*", ":P", "x3", "O_O", "xD", ":3", ";3", "^w^",
     };
 
+    private static readonly string[] defaultCensoredList =
+    {
+        "cunt", "fag", "faggot", "horny", "kook", "kys", "loli", "nigga", "nigger", "penis", "pussy", "retard", "retarded", "vagina", "vore", "restitched", "h4h"
+    }
+
+    public const string ConfigFileName = "chatCensoredList.txt";
+
+    static CensorHelper()
+    {
+
+    CensorHelper? tempConfig;
+
+    if (File.Exists(ConfigFileName) && (tempConfig = fromFile(ConfigFileName)) != null) 
+    {
+        Instance = tempConfig;
+    }
+
+    else
+    {
+        new CensorHelper().writeConfig(ConfigFileName + ".configme");
+
+        Logger.Warn
+        (
+            "The chat censor list was not found. " +
+            "A default configuration file has been created for you at " +
+            $"{Path.Combine(Environment.CurrentDirectory, ConfigFileName + ".configme")}",
+            LogArea.Config
+        );
+
+        Environment.Exit(1);
+    }
+    
+    private static INamingConvention namingConvention = CamelCaseNamingConvention.Instance;
+
+    private static CensorHelper? fromFile(string path)
+    {
+        IDeserializer deserializer = new DeserializerBuilder().WithNamingConvention(namingConvention).IgnoreUnmatchedProperties().Build();
+
+        string text;
+
+        try
+        {
+            text = File.ReadAllText(path);
+        }
+        catch
+        {
+            return null;
+        }
+
+        return deserializer.Deserialize<CensorHelper>(text);
+    }
+
+    private void writeConfig(string path)
+    {
+        ISerializer serializer = new SerializerBuilder().WithNamingConvention(namingConvention).Build();
+
+        File.WriteAllText(path, serializer.Serialize(defaultCensoredList));
+    }
+
+    }
+
```

Comment:
```suggestion
```
What you're trying to do here is basically copy the already existing configuration system, so it would make more sense to move this to a `CensorConfiguration` class in `LBPUnion.ProjectLighthouse.Configuration.ConfigurationCategories` then the filter mode and the word list could be in the same section of the lighthouse.yml config like this:
![image](https://user-images.githubusercontent.com/16675503/210679648-8c0b6430-ab94-4e2d-8079-ac6be2317c82.png)
<details>
<summary>Code</summary
</details>

```c#
public class CensorConfiguration
{
    public FilterMode UserInputFilterMode { get; set; } = FilterMode.None;

    public List<string> CensorList { get; set; } = new()
    {
        "cunt",
        "fag",
        "faggot",
        "horny",
        "kook",
        "kys",
        "loli",
        "nigga",
        "nigger",
        "penis",
        "pussy",
        "retard",
        "retarded",
        "vagina",
        "vore",
        "restitched",
        "h4h",
    };
}
```

Reply:
surely putting the word list in the lighthouse.yml is going to make it a nightmare to actually look through the config file. Maybe as the scope of our config expands we should consider making multiple config files, such as

- a listen configuration for the 3 servers
- database/redis
- monitoring
- word filters
- authentication

this way the config files become more readable and you don't have to scroll through a 400 line word filter to change the instance name (though this may be better suited for another issue)

Reply:
If we do split the configuration file then the current system needs to be refactored to more easily allow for that rather than duplicating the same class multiple times

Reply:
Alright so it sounds like I should go ahead remove this code from this commit so we can get this PR merged. Then we can open up another issue for refactoring the config?

Reply:
I have the config abstraction working in #624 but it's waiting for review